core/vm/instructions.go

```diff
@@ -518,10 +525,12 @@ func opBlockhash(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) (
 	bnum := evm.Context.BlockNumber.Uint64()
 	// if Prague is active, check if we are past the 256th block so that
 	// reading from the contract can be activated (EIP 2935).
-	if evm.chainRules.IsPrague && bnum > 256 {
-		if getBlockHashFromContract(bnum-256, evm.StateDB, evm.Accesses) != (common.Hash{}) {
+	// For testnets that start with eip2935 activated at genesis time,
+	// activate this code at block 1.
+	if evm.chainRules.IsPrague {
+		if getBlockHashFromContract(max(bnum, 256)-256, evm.StateDB, evm.Accesses) != (common.Hash{}) {
```

Comment:
instead of reading the block hash at a "wrap around" negative number, it is now assumed that if eip2935 was activated at genesis time, then we can read from the contract no matter how many blocks away from the genesis we are.

Reply:
this is a bit weird, but it's just meant to be the same thing as `max(block.number-256, 0)`

Reply:
IMHO, I'd prefer to change this to `max(block.number-256,0)` in the code, to not have to mentally deal with this difference comparing with the EIP.